lib/northbound.h

```diff
@@ -509,6 +522,17 @@ struct nb_callbacks {
 	 */
 	int (*rpc)(struct nb_cb_rpc_args *args);
 
+	/*
+	 * Notification callback.
+	 *
+	 * The callback is called when a YANG notification is received.
+	 *
+	 * args
+	 *    Refer to the documentation comments of nb_cb_notify_args for
+	 *    details.
+	 */
+	void (*notify)(struct nb_cb_notify_args *args);
```

Comment:
It's neither case. Backend clients can subscribe to receive notifications. The callback is called to handle the receipt of the notification it's associated with.

Reply:
@choppsv1  Does it mean other backend daemon wants to get notified about YANG notifications generated from another backend daemon too? For what scenario would you need to notify the YANG notfications from one backend to another? Can you please give me an example?

Reply:
Exactly. Examples? Anytime a daemon cares to know about state in some other daemon. Routes, peers, LSPs, interface state (bandwidth, loss, ...), queues, LS databases..

Reply:
@choppsv1 There already well-established channell (e.g. zebra, lib/if.h, lib/vrf.h, etc) for all these scenarios. Why do we need to notfy via MGMTD to other daemons then? All these examples you quoted are not to be treated as handling of management data. And hence these data should NOT flow through mgmtd. MGMTd's responsibility should ONLY be to notify YANG notifications from backend daemons to interested/subscribed front-end clients/sessions. We should remove the channel to notify these to other backend daemons.

Reply:
MGMTd MUST NOT be treated as a pub-sub messaging system for all FRR daemons.

Reply:
The concept is called a YANG router sometimes, it's a very common and elegant idea. To have the system state modeled by YANG and accessed by the well standardized YANG infrastructure. This larger architecture and design principle isn't something that is appropriate to debate in the comment section of this already merged PR. We have multiple different groups that all wish to use this functionality, and it's an elegant solution.

Reply:
I agree that it's better to discuss it somewhere else, and I'd like to know why exactly mgmtd **must not** be used as pub/sub router.

Reply:
Yeah edicts from "on high" not appreciated. Please raise your objections in a different forum and prepare to justify them rather than just make commandments.